src/core/layertree/qgslayertreeutils.cpp

```diff
@@ -382,6 +382,28 @@ QStringList QgsLayerTreeUtils::invisibleLayerList( QgsLayerTreeNode *node )
   return list;
 }
 
+QStringList QgsLayerTreeUtils::uncheckedGroupList( QgsLayerTreeNode *node )
+{
+  QStringList list;
+
+  if ( QgsLayerTree::isGroup( node ) )
+  {
+    const auto constChildren = QgsLayerTree::toGroup( node )->children();
+    for ( QgsLayerTreeNode *child : constChildren )
+    {
+      if ( QgsLayerTree::isGroup( child ) )
```

Comment:
I think the method should be called recursively here, so that we can have nested groups which current implementation does not consider if I am correct.

Reply:
Actually it does ! The method is called recursively on [line 400](https://github.com/qgis/QGIS/pull/41582/files#diff-1ef9764fa0bcd5fabfbf51974955fec2940df478ac3c7f27bacd70bb6d194b0eR400) with each group node.